index.js

```diff
@@ -145,82 +136,58 @@ function storeOption(longOption, optionValue, options, values) {
   }
 }
 
-const parseArgs = (config = { __proto__: null }) => {
-  const args = objectGetOwn(config, 'args') ?? getMainArgs();
-  const strict = objectGetOwn(config, 'strict') ?? true;
-  const allowPositionals = objectGetOwn(config, 'allowPositionals') ?? !strict;
-  const options = objectGetOwn(config, 'options') ?? { __proto__: null };
-
-  // Validate input configuration.
-  validateArray(args, 'args');
-  validateBoolean(strict, 'strict');
-  validateBoolean(allowPositionals, 'allowPositionals');
-  validateObject(options, 'options');
-  ArrayPrototypeForEach(
-    ObjectEntries(options),
-    ({ 0: longOption, 1: optionConfig }) => {
-      validateObject(optionConfig, `options.${longOption}`);
-
-      // type is required
-      validateUnion(objectGetOwn(optionConfig, 'type'), `options.${longOption}.type`, ['string', 'boolean']);
-
-      if (ObjectHasOwn(optionConfig, 'short')) {
-        const shortOption = optionConfig.short;
-        validateString(shortOption, `options.${longOption}.short`);
-        if (shortOption.length !== 1) {
-          throw new ERR_INVALID_ARG_VALUE(
-            `options.${longOption}.short`,
-            shortOption,
-            'must be a single character'
-          );
-        }
-      }
-
-      if (ObjectHasOwn(optionConfig, 'multiple')) {
-        validateBoolean(optionConfig.multiple, `options.${longOption}.multiple`);
-      }
-    }
-  );
-
-  const result = {
-    values: { __proto__: null },
-    positionals: []
-  };
+/**
+ * Process args and turn into identified tokens:
+ * - option (along with value, if any)
+ * - positional
+ * - option-terminator
+ *
+ * @param {string[]} args, from parseArgs({ args }) or mainArgs
+ * @param {object} options - option configs, from parseArgs({ options })
+ */
+function argsToTokens(args, options) {
+  const tokens = [];
+  let index = -1;
+  let groupCount = 0;
 
   const remainingArgs = ArrayPrototypeSlice(args);
   while (remainingArgs.length > 0) {
     const arg = ArrayPrototypeShift(remainingArgs);
     const nextArg = remainingArgs[0];
+    if (groupCount > 0)
+      groupCount--;
+    else
+      index++;
 
     // Check if `arg` is an options terminator.
     // Guideline 10 in https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html
     if (arg === '--') {
-      if (!allowPositionals && remainingArgs.length > 0) {
-        throw new ERR_PARSE_ARGS_UNEXPECTED_POSITIONAL(nextArg);
-      }
-
       // Everything after a bare '--' is considered a positional argument.
-      ArrayPrototypePushApply(
-        result.positionals,
-        remainingArgs
-      );
+      ArrayPrototypePush(tokens, { kind: 'option-terminator', index });
+      ArrayPrototypeForEach(remainingArgs, (arg) =>
+        ArrayPrototypePush(
+          tokens,
+          { kind: 'positional', index: ++index, value: arg }));
```

Comment:
```suggestion
      ArrayPrototypePushApply(tokens, ArrayPrototypeMap(remainingArgs, (arg) =>
          ({ kind: 'positional', index: ++index, value: arg })
      ));
```

Reply:
I am guessing you were viewing an older revision, as this change has already been made.

Reply:
The commit (per your earlier suggestion) was made about a day before your comment appeared. I usually push fairly soon after the commit. View Changes in this review takes me to a page with the _updated_ code. Strange.

Reply:
When I click on `index.js` above the code, I do see the unchanged code the suggestion applies to (quoted so can see references), so that at least is consistent (albeit missing the "outdated"):

`https://github.com/pkgjs/parseargs/pull/129/files/35c16f11b842e8d4690c25dbe321961489d11966#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346`

The commit with the changes is one later: `https://github.com/pkgjs/parseargs/pull/129/commits/038480af4bd50b0f54451afb16ceaf4aa5fe82b1`

Reply:
Thanks for confirming.